alibi/tests/utils.py

```diff
@@ -55,7 +56,7 @@ def __call__(self, *args, **kwargs):
         elif self.out_type == 'raw' or self.out_type == 'log_loss' or self.out_type == 'continuous':
             return self._generate_logits(sz, *args, **kwargs)
 
-    def _generate_probas(self, sz: tuple = None, *args, **kwargs) -> np.ndarray:
+    def _generate_probas(self, sz: Optional[tuple] = None, *args, **kwargs) -> np.ndarray:
```

Comment:
I think that there might be a problem with `sz` being `None` or `tuple()`:
https://github.com/SeldonIO/alibi/blob/785d2183b10a98dc6632de34023fe51f8f874571/alibi/tests/utils.py#L73
https://github.com/SeldonIO/alibi/blob/785d2183b10a98dc6632de34023fe51f8f874571/alibi/tests/utils.py#L90

I've noticed that some `numpy` functions can deal with the size being `None` or an empty tuple, but it doesn't seem to hold for all function such as `dirichlet`

Reply:
I'll see if I can validate `sz` more, although it's a minor issue for a test utility function.

Reply:
I've just checked that all the considered functions, including `dirichlet`, can take `size=None` or `size=()` as an argument, so I will leave these unchanged.